src/build-actions.js

```diff
@@ -36,63 +43,96 @@ const getWebpackConfig = async (actionPath, root, tempBuildDir, outBuildFilename
     }
     parentDir = path.dirname(parentDir)
   } while (parentDir !== rootParent && !configPath)
-  // default empty
-  const userConfig = configPath ? require(configPath) : {}
-  // needs cloning because require has a cache, so we make sure to not touch the userConfig
-  const config = cloneDeep(userConfig)
+  return configPath
+}
 
-  // entry [] must include action path
-  config.entry = config.entry || []
-  config.entry.push(actionPath)
-  config.entry = uniqueArr(config.entry)
-  // make sure filePaths are resolved from the config dir
-  config.entry = config.entry.map(f => {
-    if (!path.isAbsolute(f)) {
-      return path.resolve(path.dirname(configPath), f)
-    }
-    return f
-  })
+/**
+ *  Loads a Webpack config file from the config path provided. Sets fields required
+ *  for Runtime actions. Returns an object that can be passed to the Webpack library.
+ *
+ * @param {string} configPath Path of the Webpack config file
+ * @param {string} actionPath Path of the action
+ * @param {string} tempBuildDir Path of the output directory for the bundle
+ * @param {string} outBuildFilename Name of the output file for the action
+ * @returns {Promise<object>} Webpack config, can be passed to the Webpack library
+ */
+const loadWebpackConfig = async (configPath, actionPath, tempBuildDir, outBuildFilename) => {
+  const configs = []
+  const cliEnv = getCliEnv()
+  let importConfig = configPath ? require(configPath) : {}
 
-  // if output exists, default to commonjs2
-  config.output = config.output || {}
-  if (config.output.libraryTarget === undefined) {
-    config.output.libraryTarget = 'commonjs2'
+  if (typeof importConfig === 'function') {
+    importConfig = await importConfig(process.env)
   }
```

Comment:
```suggestion
  if (typeof importConfig === 'function') {
    importConfig = await importConfig(process.env)
  } else {
    // If the config file exported a Promise, resolve the Promise.
    // Otherwise this is a no-op.
    importConfig = await importConfig;
  }
```

The one Webpack-supported config format this doesn't support yet is [a file that exports a Promise directly](https://webpack.js.org/configuration/configuration-types/#exporting-a-promise), rather than a Promise-returning function. This change could fix that.